openxr/src/graphics/headless.rs

```diff
@@ -0,0 +1,68 @@
+use crate::*;
+
+/// Used to create a session without graphics.
+///
+/// See [`XR_MND_headless`] for details.
+///
+/// [`XR_MND_headless`]: https://www.khronos.org/registry/OpenXR/specs/1.1/html/xrspec.html#XR_MND_headless
+pub enum Headless {}
+
+impl Graphics for Headless {
+    type Requirements = Requirements;
+    type SessionCreateInfo = SessionCreateInfo;
+    type SwapchainImage = HeadlessSwapchainImage;
+    type Format = HeadlessFormat;
+
+    fn raise_format(_: i64) -> Self::Format {
+        // used by enumerate_swapchain_formats, which returns empty
+        unreachable!()
```

Comment:
nit: This is technically reachable by user code, but probably indicates a bug. Maybe `panic!` with a helpful message?

Reply:
one would need both of these to be true in order to hit this:
- the openxr runtime is not spec compliant and returns a non-empty set of formats from `xrEnumerateSwapchainFormats`
- the developer calls `enumerate_swapchain_formats`, disregarding headless session limitations

if the runtime is to spec, `xrEnumerateSwapchainFormats` will return an empty set and the inside of the `.map` that calls `raise_format` will never get executed.

Reply:
It's a public method, someone can just call it directly. Not really important since it's hidden, though.